### ISSUE TITLE
Type Error - awsService[method] is not a function

### ISSUE THREAD

**Issue by krolique** [Jimdo/serverless-dynamodb-ttl#14]
Foremost I thank you kindly for writing this plugin as Cloudformation does not offer the option of automating the TTL features on a `DynamoDB` resource. 

I followed the tutorial step by step, installed the plugin via `npm`, added the plugin in the `serverless.yml` and then added the `customs`  with the section in the config. 

However then I attempt to deploy with `sls deploy -s dev` (my sls setup requires a stage flag)  
I get the following error:

```
Serverless: Enabling TTL setting(s) for DynamoDB (us-east-1)
 
  Type Error ---------------------------------------------
 
     awsService[method] is not a function
 
     For debugging logs, run again after setting the "SLS_DEBUG=*" environment variable.
 
  Stack Trace --------------------------------------------
 
TypeError: awsService[method] is not a function
    at persistentRequest (/usr/local/lib/node_modules/serverless/lib/plugins/aws/provider/awsProvider.js:146:37)
    at doCall (/usr/local/lib/node_modules/serverless/lib/plugins/aws/provider/awsProvider.js:130:9)
    at BbPromise (/usr/local/lib/node_modules/serverless/lib/plugins/aws/provider/awsProvider.js:141:14)
    at persistentRequest (/usr/local/lib/node_modules/serverless/lib/plugins/aws/provider/awsProvider.js:128:38)
    at AwsProvider.request (/usr/local/lib/node_modules/serverless/lib/plugins/aws/provider/awsProvider.js:144:12)
    at Plugin.check (/REDACTED/node_modules/serverless-dynamodb-ttl/src/plugin.js:46:26)
    at list.map.data (/REDACTED/node_modules/serverless-dynamodb-ttl/src/plugin.js:36:22)
    at Array.map (native)
    at Promise.resolve.then.then.then (/REDACTED/node_modules/serverless-dynamodb-ttl/src/plugin.js:35:25)
    at process._tickDomainCallback (internal/process/next_tick.js:135:7)
 
  Get Support --------------------------------------------
     Docs:          docs.serverless.com
     Bugs:          github.com/serverless/serverless/issues
     Forums:        forum.serverless.com
     Chat:          gitter.im/serverless/serverless
 
  Your Environment Information -----------------------------
     OS:                 darwin
     Node Version:       7.10.0
     Serverless Version: 1.13.0
```

This is a sub section of my `serverless.yml` file that contains the definition for the table and TTL

```
resources:

    Resources:
      MyTableResource:
       Type: "AWS::DynamoDB::Table"
       DeletionPolicy: Retain
       Properties:
         TableName: MyTable
         AttributeDefinitions:
           - AttributeName: token
             AttributeType: S
         KeySchema:
           - AttributeName: token
             KeyType: HASH
         ProvisionedThroughput:
           ReadCapacityUnits: 10
           WriteCapacityUnits: 10

# External Variables
custom:
  vars: ${file(../vars.yml)}

  dynamodb:
    ttl:
      - table: MyTable
        field: ttl
```

**Comment by sbstjn** [comment]
Sorry for running into this issue. I'll have a look into it …

**Comment by sbstjn** [comment]
@krolique could you update to a more recent `serverless` (`1.15.3`) version and try it again? I just tested an outdated version and it failed with the same error, but it works fine when using the most recent version.

* `1.13.0` fails
* `1.13.2` works fine
* `1.15.x` works fine

**Comment by sbstjn** [comment]
I created a PR to check the used serverless version. The plugin will then show a warning and you should not run into this error anymore. But you should update your serverless version anyway ;)

**Comment by sbstjn** [comment]
DynamoDB TTL settings are included in `aws-sdk` since version `2.21`.

```
serverless 1.15.3
 => aws-sdk 2.71.0
 
serverless 1.13.2
 => aws-sdk 2.50.0
 
serverless 1.13.0
 => aws-sdk 2.7.13
```

**Comment by sbstjn** [comment]
The check is included in the most recent `0.1.1` version. It should be available using npm in a couple of minutes (deployment just started …).

**Comment by krolique** [comment]
@sbstjn Thank you so much for addressing my concerns. I spend this morning trying to switch my setup to use `serverless` versions `1.13.1`, `1.13.2` and `1.15.3` however because I use `${self}.custom.SOMETHING` and the following bug https://github.com/serverless/serverless/issues/3627 I am unable to upgrade to either of these versions.

I've checked their release log and issue 3627 has not been fixed. Sadly I am going to have to wait until the fix is in place to start using this library 😢
